lib/matplotlib/colorbar.py

```diff
@@ -294,6 +228,70 @@ def draw(self, renderer):
         return ret
 
 
+class _ColorbarAxesLocator:
+    """
+    Wrap any locator on the axes, or its position, to shrink the
+    inner axes if there are extend triangles at either min or max.
+    """
+    def __init__(self, cbar):
+        """
+        *bounds* (a ``[l, b, w, h]`` rectangle) and *transform* together
+        specify the position of the inset axes.
+        """
+        self._cbar = cbar
+        self._orig_locator = cbar.ax._axes_locator
+
+    def __call__(self, ax, renderer):
+
+        # make sure that lims and scales are the same
+        scale = self._cbar._long_axis().get_scale()
+        try:
+            self._cbar._short_axis()._set_scale(scale)
+        except TypeError:
```

Comment:
Yeah, this is a bit of a pickle - if the scale requires an argument like function scale, then `_set_scale(scale)` requires the optional arguments.  But we lost those, so that fails.  So, more complicated scales will always fail here.  

Note this is only needed if we change the scale after the colorbar is made.

Reply:
Removed